lib/provisioning/Provisioner.js

```diff
@@ -0,0 +1,112 @@
+/*eslint no-invalid-this: 0*/ // eslint doesn't understand Promise.coroutine wrapping
+"use strict";
+var Promise = require("bluebird");
+var IrcRoom = require("../models/IrcRoom");
+var MatrixRoom = require("matrix-appservice-bridge").MatrixRoom;
+
+var log = require("../logging").get("Provisioner");
+
+function Provisioner(ircBridge) {
+    this._ircBridge = ircBridge;
+
+    let as = this._ircBridge.getAppServiceBridge().appService;
+
+    as.app.post("/_matrix/provision/link", (req, res) => {
+        return this.validate(req.body).then(
+            () => { res.json({}) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        ).then(
+            () => { return this.link(req.body) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        );
+    });
+
+    as.app.post("/_matrix/provision/unlink", (req, res) => {
+        return this.validate(req.body).then(
+            () => { res.json({}) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        ).then(
+            () => { return this.unlink(req.body) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        );
+    });
+}
+
+var parameterValidation = {
+    matrix_room_id :
+        {regex : /^![0-9A-Za-z]+?:[^:]+(\:[0-9]+)?$/, example : '!Abcdefg:example.com[:8080]'},
+    remote_room_channel :
+        {regex : /^#.+$/, example : '#roomname'},
+    remote_room_server :
+        {regex :  /^[a-z\.]+$/, example : 'example.com or localhost'},
```

Comment:
You need to be more lax than this. IP addresses for example won't work, nor will punycode domain names. Refer to the [RFC](https://tools.ietf.org/html/rfc3986#appendix-A) for the `host` BNF.